determined-ai/determined
# feat: `det deploy --image-repo-prefix` for pulling images from a custom docker repo

## Description

This new option `det deploy --image-repo-prefix` is aimed at contributors, and allows to switch images used for the master and agent nodes from `determinedai/{determined-master,determined-agent}` to `CUSTOM_REPO/{determined-master,determined-agent}`. It can be useful for developers who want to build local docker images for these services, push them to their personal dockerhub repository (or perhaps other registry), and then do an end-to-end deploy of an aws or gcp cluster. 
This option also works for `det deploy local`. Pushing the images to the remote registry is not required for this case.

## Test Plan

1. Build master & agent images.
2. Tag and push them to your personal dockerhub repo
3. Run `det deploy --image-repo-prefix determinedilia aws up ...`
4. Make sure the cluster runs the custom images (and not the ones from `determinedai` repo).

## Commentary (optional)




## Checklist

See [Release Note](../docs/release-notes/README.md) for details.
